solutions/3653/01.py

```
@@ -1,16 +1,50 @@
-        for l, r, k, v in queries:
-            idx = l
-            while idx <= r:
-                nums[idx] = (nums[idx] * v) % MOD
-                idx += k
-
-        # Calculate XOR of all elements
-        res = 0
-        for num in nums:
-            res ^= num
-        return res
+    def countPaths(self, n: int, edges: List[List[int]]) -> int:
```

Ruff (0.14.10)

Undefined name `List`

(F821)